ios_ui/MovieRec/RecommendationsTableViewController.swift

```diff
@@ -66,23 +70,62 @@ class RecommendationsTableViewController: UITableViewController {
         makeAPICall()
     }
     
-    private func refreshMovies(data: Data) -> Void {
-        let responseJSON = try? JSONSerialization.jsonObject(with: data, options: [])
-        self.recommendations.removeAll()
-        if let responseJSON = responseJSON as? [String: [String]] {
-            for recommendation in responseJSON["recommendations"]! {
-                self.recommendations += [Recommendation(title: recommendation)]
-            }
+    private func refreshMovies(data: JSON) -> Void {
+        for recommendation in data.arrayValue {
+            self.recommendations += [Recommendation(title: recommendation.stringValue)]
+        }
 
-            DispatchQueue.main.async {
-                self.tableView.reloadData()
-            }
+        DispatchQueue.main.async {
+            self.tableView.reloadData()
         }
+        
         self.ratings.removeAll()
         delegate?.clearRatings()
         print(ratings.count)
     }
     
+    private func startJobPolling(data: Data) -> Void {
+        let responseJSON = try? JSONSerialization.jsonObject(with: data, options: [])
+        if let responseJSON = responseJSON as? [String: String] {
+            print(responseJSON["job_id"]!)
+            startTimer(job_id: responseJSON["job_id"]!)
+        }
+    }
+    
+    private func checkJobStatus(data: Data) -> Void {
+        let json = JSON(data: data)
+        if json["status"].stringValue == "completed" {
+            print("job IS completed")
+            if let dataFromString = json["results"].stringValue.data(using: .utf8, allowLossyConversion: false) {
+                let results = JSON(data: dataFromString)
+                refreshMovies(data: results)
+            }
+            stopTimer()
+        } else {
+            print("job not completed")
+        }
+    }
+    
+    
+    func startTimer(job_id: String) {
+        let queue = DispatchQueue(label: "com.domain.app.timer")
+        timer = DispatchSource.makeTimerSource(queue: queue)
+        timer!.scheduleRepeating(deadline: .now(), interval: .seconds(5))
```

Comment:
I wouldn't force unwrap this timer object. I would use guard let timer = DispatchSource.makeTimerSource(queue: queue) else { fatalError("timer was not created") }

you can decide if this should fatal out or not, otherwise inside the else block you can just return. and maybe log something out indicating it messed up